### PR TITLE
introduce `property_checker_resultt`

### PR DESCRIPTION
**Pull request by kroening**
This introduces a type for the result from an engine, as a replacement for `int`.

### CODE REVIEW

File: src/ebmc/property_checker.h [hunk 1]
```diff
@@ -12,6 +12,13 @@ Author: Daniel Kroening, dkr@amazon.com
 #include "ebmc_properties.h"
 #include "transition_system.h"
 
+enum class property_checker_resultt
+{
+  REPORT_RESULT,
```
Comment (tautschnig) [inline src/ebmc/property_checker.h:17]: I am struggling to understand what this one is about? Or, rather, why is "SUCCESS" not a "result?"

**Comment by tautschnig** [comment]
Also, is there a case for re-using `src/goto-checker/properties.h`?

File: src/ebmc/property_checker.h (hunk 1 above)
Comment (kroening) [reply src/ebmc/property_checker.h:17]: It's not a verification result -- rename REPORT_RESULT -> REPORT_VERIFICATION_RESULT?

**Comment by kroening** [comment]
> Also, is there a case for re-using `src/goto-checker/properties.h`?

It will move into this direction; but there simply aren't any goto programs.

File: src/ebmc/property_checker.h (hunk 1 above)
Comment (kroening) [reply src/ebmc/property_checker.h:17]: Or simply `VERIFICATION_RESULT`?

Comment (tautschnig) [reply src/ebmc/property_checker.h:17]: `VERIFICATION_RESULT` works for me.

Comment (kroening) [reply src/ebmc/property_checker.h]: done

**Review by tautschnig — APPROVED** [review]

**Pull request merged**